web/src/main/java/org/springframework/security/web/jackson2/WebJackson2Module.java

```diff
@@ -53,12 +54,14 @@ public WebJackson2Module() {
 
 	@Override
 	public void setupModule(SetupContext context) {
-		SecurityJackson2Modules.enableDefaultTyping((ObjectMapper) context.getOwner());
-		context.setMixInAnnotations(Cookie.class, CookieMixin.class);
-		context.setMixInAnnotations(SavedCookie.class, SavedCookieMixin.class);
+		SecurityJackson2Modules.enableDefaultTyping(context.getOwner());
+		if (ClassUtils.isPresent("javax.servlet.http.Cookie", this.getClass().getClassLoader())) {
+			context.setMixInAnnotations(Cookie.class, CookieMixin.class);
```

Comment:
Depending on the JDK this can cause issues. The reason is that many JDK's will load all the classes necessary for the entire class to run at once even if the code is not necessarily going to be executed. Instead, this code needs to be isolated in a separate module and then conditionally loaded.

Reply:
Hm, that's interesting, I didn't know that..
Is that means that some JDK(if it's fast could you give me example of that specific JDK or link to specification?) will try to load all the classes that presents in that class in initialization step of the that class? And that 'Cookie.class' will be try to load without looking at the condition?
But if I put everything in brackets into a separate class(new module) that could be load but will not instantiate, base on condition, that will work? 
And that loading of all necessary classes will done only if I will try to create instance of that new module?